### ISSUE TITLE
Discord session silently parks on Claude folder-trust prompt after CUSTOMER_HOME migration

### ISSUE THREAD

**Issue by jacketyjax** [scrollinondubs/behalfbot#21]
## Summary

Discord session silently parks on Claude Code's first-run folder-trust dialog after the `#6` CUSTOMER_HOME migration, on any install whose CUSTOMER_HOME path has no trust entry in `~/.claude.json`. Bot appears alive (tmux session exists, process is healthy) but never connects to Discord. Watchdog does not detect it.

Reported and root-caused by @tobyzn on his `asimov` install: full writeup at https://gist.github.com/tobyzn/2e70b548e94d497b18f71b7ac6c9f407 - this issue lifts his analysis upstream so the fix lands in chassis for every install going forward.

## Symptom

- Discord bot stops replying. No crash, no error logs.
- `tmux ls` shows the `*-discord` session is alive.
- `scripts/` directory is intact (so this is **not** the 2026-06-02 "re-bootstrap deleted scripts" incident from `#new-jaxity#XXX`).
- `tmux capture-pane -t <session> -p | tail -30` reveals the claude process is parked on:

  ```
  Accessing workspace: /Users/<user>/.behalfbot
  Is this a project you trust?
   > 1. Yes, I trust this folder
     2. No, exit
  ```

## Root cause

1. Claude Code stores folder-trust state **per absolute path** in `~/.claude.json` under `projects["<abs path>"].hasTrustDialogAccepted`.
2. `--dangerously-skip-permissions` does **not** bypass the folder-trust gate. It controls the per-tool permission classifier, which is a separate downstream check that only fires after trust is established.
3. The `#6` CUSTOMER_HOME migration changed the Discord launch cwd from CHASSIS_HOME (`~/behalfbot`, already trusted on pre-migration installs) to CUSTOMER_HOME (`~/.behalfbot`, **never trusted** on any install).
4. Result: every Discord restart on a post-migration install boots Claude into a path with no trust entry, hits the interactive prompt, and waits for keyboard input forever.

### Secondary contradiction in restart template

`chassis/scripts/templates/restart-discord.sh.template` has a comment claiming the working dir is CHASSIS_HOME:

> Working dir is CHASSIS_HOME so claude resolves CLAUDE.md from there.

…but the rendered command actually launches from CUSTOMER_HOME:

```bash
tmux new-session -s "<name>-discord" -d \
    "cd '<CUSTOMER_HOME>' && claude --channels 'plugin:discord@claude-plugins-official' --dangerously-skip-permissions"
```

Comment and behavior disagree. Even after the trust fix, this should be reconciled.

## Why the watchdog missed it

`com.jax.discord-watchdog` (and equivalent watchdogs on other installs) checks for the existence of the `*-discord` tmux session. The session **was** alive - it just had a process parked on an interactive prompt that nothing was around to answer.

A liveness check on the tmux session is not a liveness check on "is the bot actually listening to Discord." This is the **second** install hit by a "tmux alive, bot dead" failure shape - the first was the post-bootstrap missing-scripts incident. The watchdog should be hardened against this class of bug regardless of which specific prompt is parking the process.

### 1. Pre-seed trust at bootstrap (kills failure mode at source)

When bootstrap establishes CUSTOMER_HOME, idempotently merge a trust entry for the path into `~/.claude.json`:

```json
{
  "projects": {
    "<CUSTOMER_HOME>": {
      "hasTrustDialogAccepted": true,
      "hasCompletedProjectOnboarding": true
    }
  }
}
```

This eliminates the failure mode for every future install, not just this one symptom. Must be idempotent because users may run bootstrap multiple times.

### 2. Watchdog checks pane content, not just session existence

Have the watchdog parse `tmux capture-pane -t <session> -p` output and confirm the pane has reached the:

```
Listening for channel messages from: plugin:discord@claude-plugins-official
```

state. If the session is alive but the pane has not progressed to "Listening" within N seconds of start, treat it as dead and restart.

This catches **any** future variant of "tmux alive but bot parked on a prompt," not just this trust-dialog case. Worth doing even after fix #1 lands - belt and suspenders.

### 3. Fix the cwd/comment contradiction in the restart template

Either:
- Change the `cd` target to CHASSIS_HOME and keep the comment accurate, or
- Keep the `cd` to CUSTOMER_HOME (post-fix #1 this is safe) and update the comment to match.

Don't ship a template where the comment lies about what the code does. Cleanup, not blocking.

### 4. Document the trust step in the migration runbook

Minimum bar regardless of the above: add a section to the migration/runbook calling out folder-trust as a thing that has to be re-established when launching from a new path. Defense in depth in case a future migration introduces a similar issue.

## Fast diagnostic for this class of bug

If a bot is "alive" (tmux session present) but silent:

```bash
tmux capture-pane -t <session-name> -p | tail -30
```

If you see an interactive prompt rather than "Listening for channel messages," that's this bug or a sibling.

## Affected installs

- @tobyzn / `asimov` (confirmed, recovered via manual `~/.claude.json` edit + restart)
- Any other install that bootstrapped post-`#6` whose CUSTOMER_HOME differs from CHASSIS_HOME and was not previously trust-accepted. Sean's install was not affected because his CUSTOMER_HOME path predates the migration and is already trusted.

## Credit

Root cause analysis, reproduction, manual recovery procedure, and suggested fixes by @tobyzn. Issue filed upstream by Jax on Sean's behalf - this is Toby's bug report verbatim where it can be acted on.

Gist: https://gist.github.com/tobyzn/2e70b548e94d497b18f71b7ac6c9f407

**Comment by jacketyjax** [comment]
PRs up - splitting the fix into two so each can land and roll back independently:

- **#22** — bootstrap pre-seed of `~/.claude.json` trust entry for CUSTOMER_HOME. Causal fix; eliminates the failure mode on every future install.
- **#23** — watchdog hardening to detect the trust prompt (and any future TUI prompt) on pane content. Defense in depth; catches "tmux alive but bot parked" variants we haven't seen yet.

@tobyzn ready to test on `asimov` when you are. Recommended order: merge #22 first, then #23 - that way the watchdog's diagnostic log message can usefully point at the bootstrap pre-seed as the upstream fix. Happy to share either branch with you for pre-merge testing.

**Comment by jacketyjax** [comment]
Both fixes merged:

- #22 — bootstrap pre-seed of `~/.claude.json` trust entry for CUSTOMER_HOME (causal fix)
- #23 — watchdog detection of folder-trust prompt + generic TUI menu parking (defense in depth)

Operators on existing installs: re-run `bash chassis/scripts/bootstrap-customer-scripts.sh` from CHASSIS_HOME to pick up both the trust pre-seed and the re-rendered watchdog script. New installs from `main` get both automatically.

Credit to @tobyzn for the report, root cause, and recovery procedure.
